README.md

```diff
@@ -1,196 +1,32 @@
 
-# Git Clang-Format will read properties from your git configuration.  You can set default properties
-# in your git configuration or pass them to the command directly.  This script assumes you'll set options
-# in your git configuration.  Git Clang-Format configuration options include:
-#   clangformat.binary
-#   clangformat.style
-#   clangformat.extensions
-#
-# You'd set them by running the following commands:
-#   git config --global clangformat.binary clang-format
-#   git config --global clangformat.style file
-#   git config --global clangformat.extensions 'h,c,hpp,tpp,cpp'
-#
-# You can pass them directly by using the line below (if you do this, you'll need to pass them everywhere):
-#   cmd = ['git', 'clang-format', '--extensions', 'h,c,hpp,tpp,cpp', '--style', 'file', '--diff']
-#   output = check_output(cmd)
-output = check_output(['git', 'clang-format', '--diff'])
+### Adding a Git Pre-Commit Hook
+Git can run clang-format against your changes before being committed. This guarantees that the formatting of your code always matches the requirements of the project. A pre-commit hook is provided by this repository.
 
-messages = [
-    'no modified files to format\n',
-    'clang-format did not modify any files\n'
-]
-
-# If you receive this message and you have unstaged changes, you may need to run the following to
-# allow changes to those files as well.
-#   git clang-format -f
-if output not in messages:
-    print('Run git clang-format [-f], then commit.\n')
-    exit(1)
-
-exit(0)
+Before you can use the pre-commit hook, you'll need to make sure your git configuration defines the correct properties. To do that, run the following commands:
+```bash
+git config --global clangformat.binary clang-format
+git config --global clangformat.style file
+git config --global clangformat.extensions 'h,c,hpp,cpp'
 ```
-To test, modify a C/C++ file and commit the changes.&nbsp; You should receive a message stating that you need to run
-`git clang-format`.&nbsp; Do just that and you should be good to go.&nbsp; The tool will only format the modified code.
-
-### Notes
-- If you change the configuration, make sure to undo any formatting previously done.&nbsp; Not doing so could result in unexpected results.
-- Use `git clang-format` instead of `clang-format` so that only the modified files will be formatted.
 
-### Formatting Files Outside Of Git
-**WARNING!  THESE COMMANDS TOUCH MULTIPLE FILES!**
-The following commands assume you are at the root of the project and using the BaSH shell.
-These commands are not meant to be run together (run only one).
-Use the pre-commit hook after using these going forward.
+To enable the pre-commit script, do the following:
+```bash
+cd root/of/your/repo
+cd .git/hooks
+cp /path/to/irods_clang_format_configuration/pre-commit.template pre-commit
```

Comment:
possibly combine these lines...

```
cp /path/to/irods_clang_format_configuration/pre-commit.template ./.git/hooks/pre-commit
```

Reply:
oh, then the chmod line would have to change too...  hmmm, yeah, either update the chmod too - or leave all as is.

Reply:
There's a chmod that follows.